apps/src/sites/studio/pages/code-studio.js

```diff
@@ -105,3 +108,20 @@ window.CDOSounds = Sounds.getSingleton();
 
 checkForUnsupportedBrowsersOnLoad();
 initHamburger();
+
+$(document).ready(() => {
+  // Attempt to replicate logic used by user_header.haml to populate the name
+  // in our Sign In button.
+  if (window.userNameCookieKey) {
+    const val = cookies.get(window.cookieKey);
+    if (val) {
+      store.dispatch(setUserSignedIn(true));
+    } else {
+      // We did not have a cookie, meaning we're probably not signed in. For scripts
+      // that are not cached, the server will set the user id in the DOM if you're
+      // signed in so check that as well.
+      const nameSpan = document.querySelector('.header_button.header_user.user_menu .user_name');
+      store.dispatch(setUserSignedIn(nameSpan && nameSpan.dataset.id));
```

Comment:
I think the only way this happens is if the user deletes their cookies. In that case, we'll still fill in their user name correctly on non-cached scripts because the server set dataset.id. We should be as resilient as the header button.

Reply:
I don't understand how this works.  Does Rails have a way to identify the user's session that doesn't depend on cookies?

Reply:
Good question. I only deleted the _shortName cookie. I think there's another session cookie that probably still has user info (and allows the server to reconstruct the _shortName cookie). That does mean we probably don't really need to care about this scenario since people are probably deleting all of their cookies (in which case server shouldnt know who you are) or none of them.